meerk40t/gui/panes/navigationpanels.py

```diff
@@ -998,34 +1021,33 @@ def on_reset(self, event=None):  # wxGlade: Navigation.<event_handler>
         self.matrix_updated()
 
     def on_rotate_ccw(self, event=None):  # wxGlade: Navigation.<event_handler>
-        spooler, input_driver, output = self.context.registered[
-            "device/%s" % self.context.root.active
-        ]
-        x = input_driver.current_x if input_driver is not None else 0
-        y = input_driver.current_y if input_driver is not None else 0
-        self.context("rotate %fdeg %f %f\n" % (-5, x, y))
-        self.matrix_updated()
+        angle = -5.0
+        self.rotateit(angle)
 
     def on_rotate_cw(self, event=None):  # wxGlade: Navigation.<event_handler>
-        spooler, input_driver, output = self.context.registered[
-            "device/%s" % self.context.root.active
-        ]
-        x = input_driver.current_x if input_driver is not None else 0
-        y = input_driver.current_y if input_driver is not None else 0
-        self.context("rotate %fdeg %f %f\n" % (5, x, y))
-        self.matrix_updated()
+        angle = 5.0
+        self.rotateit(angle)
+
+    def on_rotate_ccw_90(self, event=None):  # wxGlade: Navigation.<event_handler>
+        angle = -90.0
+        self.rotateit(angle)
+
+    def on_rotate_cw_90(self, event=None):  # wxGlade: Navigation.<event_handler>
+        angle = 90.0
+        self.rotateit(angle)
 
     def on_text_matrix(self, event=None):  # wxGlade: Navigation.<event_handler>
         try:
+            # Not yet working as kernel reports a syntax error...
```

Comment:
Can you comment out lines 4107, 4130-4131 in elements.py and try to see what is creating a ValueError which is propagated as a SyntaxError?

Reply:
See [PR#824](https://github.com/meerk40t/meerk40t/pull/824) - I was splitting these two PRs intentionally, as this PR is more of a cometic nature (although it fixes some nasty bugs as well, like the transformation matrix not updated after dragging / resizing etc.). If you want me to combine these then let me know, as this PR is rather useless without the aforementioned.